### PR TITLE
Fix handling of byte arrays

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
Byte arrays in URL format require special marshallers and unmarshallers.
Consolidate duplicate code for JSON marshaller and internal marshallers.
Fixed scope for azkeyvault.

### CODE REVIEW

File: src/generator/models.ts [hunk 1]
```diff
@@ -62,6 +66,17 @@ export async function generateModels(session: Session<CodeModel>): Promise<strin
     text += '\t}\n';
     text += '}\n\n';
   }
+  if (needsJSONPopulateByteArray) {
+    text += 'func populateByteArray(m map[string]interface{}, k string, b []byte, f azcore.Base64Encoding) {\n';
+    text += '\tif len(b) == 0 {\n';
+    text += '\t\treturn\n';
+    text += '\t} else if azcore.IsNullValue(b) {\n';
+    text += '\t\tm[k] = nil\n';
+    text += '\t} else if !reflect.ValueOf(b).IsNil() {\n';
```
Comment (catalinaperalta) [inline src/generator/models.ts:75]: Just checking do we need to do the IsNullValue and IsNil with those azcore methods for a byte array just like we did for the interfaces?

Comment (jhendrixMSFT) [reply src/generator/models.ts:75]: D'oh I think you're right, this can be simplified.  I also need to test the JSON `null` scenario too.

Comment (jhendrixMSFT) [reply src/generator/models.ts]: Fixed with test.  We do need the `IsNullValue()` to handle the JSON null scenario.

**Review by catalinaperalta — APPROVED** [review]
LGTM
